.github/workflows/tests.yaml

```diff
@@ -11,18 +11,51 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v2
+
       - uses: actions/setup-node@v2
         with:
           node-version: '16'
+
       - uses: actions/cache@v2
         with:
           path: '**/node_modules'
           key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
       - name: install dependencies
         run: yarn install --immutable --network-concurrency=1
+
       - name: run lint
         run: yarn run --silent lint -- --max-warnings 0
+
       - name: set host bucketclient.testing.local
         run: sudo bash -c 'echo "127.0.0.1 bucketclient.testing.local" >> /etc/hosts'
+
       - name: run test
         run: yarn run --silent test
+
+  go-client-test:
+    name: Test the Go client
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        shell: bash
+        working-directory: go-client/
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.23'
+
+      - name: Install ginkgo and golangci-lint
+        run: |
+          go install github.com/onsi/ginkgo/v2/ginkgo@v2.20.2
+          go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.61.0
+
+      - name: Run go linter
+        run: make lint
+
+      - name: Run unit tests
+        run: make test-coverage
```

Comment:
we probably want to take the change to activate `codecov` already, so we don't accumulate debt while building up the project (c.f. https://github.com/scality/sorbet/blob/HEAD/codecov.yml and https://github.com/scality/sorbet/blob/HEAD/.github/workflows/tests.yaml#L119-L126 for exemple)

Reply:
Will do next week, separately from this PR (https://scality.atlassian.net/browse/BKTCLT-20). I made a PR for metadata-migration: https://github.com/scality/metadata-migration/pull/8